src/HttpCollection.ts

```diff
@@ -0,0 +1,37 @@
+import {Action} from '@/providers/Provider';
+import Collection from '@/Collection';
+import HasHttpProvider from '@/providers/HasHttpProvider';
+import HttpProvider from '@/providers/HttpProvider';
+
+export default abstract class HttpCollection<M, A> extends Collection<M, A> implements HasHttpProvider {
+    /**
+     * @inheritdoc
+     */
+    abstract endpoint: string;
+
+    /**
+     * @inheritdoc
+     */
+    provider = new HttpProvider();
+
+    /**
+     * @inheritdoc
+     */
+    getEndpoint(action?: Action): string {
```

Comment:
```suggestion
    getEndpoint(): string {
```

Should `action` be there since collection only has `fetch`?

Reply:
It does not only have `fetch`. People can use it in any way they want. But I will indeed remove it. Since it is not used in this method.